olp-cpp-sdk-core/src/cache/DefaultCache.cpp

```diff
@@ -49,6 +51,21 @@ time_t GetRemainingExpiryTime(const std::string& key,
   return expiry;
 }
 
+/// Return success flag, cache data and remaing expiry time
+std::tuple<bool, DataPtr, time_t> GetFromCache(
```

Comment:
We should have something like optional<Result>, result = data + expiry

Reply:
Why do you need the bool for? shared_ptr == nullptr is not enough?

Reply:
Redone.

Reply:
I've added a bool to ease the if conditions in Get method. Otherwise, we will need to checks both the data and the expiry. Though, it is not relevant anymore.